multinet/uploaders/csv.py

```diff
@@ -47,9 +49,25 @@ def validate_csv(rows):
                 detail.append({"fields": fields, "row": i + 2})
 
         if detail:
-            return {"error": "syntax", "detail": detail}
+            data_errors.append({"error": "syntax", "detail": detail})
+    else:
+        # Unsupported Table, error since we don't know what's coming in
+        data_errors.append({"error": "unsupported"})
+
+    if len(data_errors) > 0:
+        raise ValidationFailed(data_errors)
+    else:
+        return None
 
-    return None
+
+def decode_data(input):
+    """Decode the request data assuming utf8 encoding."""
+    try:
+        body = input.decode("utf8")
+    except UnicodeDecodeError:
+        raise DecodeFailed([{"error": "unsupported", "detail": "not utf8"}])
```

Comment:
Updated in the next commit

Reply:
I agree that this is probably best for this one function but since ValidationFailed returns `{"errors": errors}` maybe we should be consistent? This would allow us to do just one check from the client.

Reply:
The problem is that ValidationError can return multiple errors. How should we update that to handle the multiple errors?